### PR TITLE
[sw, dif_plic] Prepare dif_plic for S1 signoff

### PR DESCRIPTION
**Pull request by silvestrst**
This change prepares dif_plic for S1 signeoff. Please refer to the [checklist document](https://docs.opentitan.org/doc/project/checklist/#s1) for S1 signeoff requirements.

### CODE REVIEW

**Comment by lenary** [comment]
You will also need to update:
- [ ] `hw/top_earlgrey/dv/chip_sim_cfg.hjson`
- [ ] `ci/run_verilator_pytest.sh`

**Comment by silvestrst** [comment]
> You will also need to update:
> 
> * [x]  `hw/top_earlgrey/dv/chip_sim_cfg.hjson`
> * [x]  `ci/run_verilator_pytest.sh`
> 
> Edit: And re-run `clang-format` because `git-clang-format` is slightly dumb with file moves.

Thanks!

**Comment by imphil** [comment]
If the consecutive_irqs_test is not tested any more think about removing it completely.

**Comment by lenary** [comment]
@imphil it should still be tested, we've just moved the source file for it.

**Comment by imphil** [comment]
Ah, I missed the move in the diff view. Sorry for the noise.

File: ci/run_verilator_pytest.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ BOOT_ROM_TARGET="boot_rom/boot_rom_sim_verilator.elf"
 TEST_TARGETS=(
   "examples/hello_usbdev/hello_usbdev_sim_verilator.elf"
   "tests/aes_test_sim_verilator.elf"
-  "tests/consecutive_irqs_test_sim_verilator.elf"
+  "tests/dif_plic_sanitytest.elf"
```
Comment (lenary) [inline ci/run_verilator_pytest.sh:20]:
```suggestion
  "tests/dif_plic_sanitytest_sim_verilator.elf"
```

**Comment by lenary** [comment]
I have a slight change to the meson for you - it's better to put the build of `dif_plic_sanitytest` in the same directory as the test, and you can add to `sw_tests` directly from there (due to how the meson is written): 
[plic-move-meson-changes.diff](https://github.com/lowRISC/opentitan/files/4745863/plic-move-meson-changes.diff.txt)

**Comment by silvestrst** [comment]
> it's better to put the build of dif_plic_sanitytest in the same directory as the test

Good point, I am on it!

File: hw/top_earlgrey/dv/chip_sim_cfg.hjson [hunk 2]
```diff
@@ -109,9 +109,9 @@
       sw_test: sw/device/tests/aes_test
     }
     {
-      name: chip_consecutive_irqs_test
+      name: chip_dif_plic_sanitytest
       uvm_test_seq: chip_sw_base_vseq
-      sw_test: sw/device/tests/consecutive_irqs_test
+      sw_test: sw/device/tests/dif/dif_plic_sanitytest
```
Comment (sriyerg) [inline hw/top_earlgrey/dv/chip_sim_cfg.hjson:114]: This probably does not work. The `dif` directory should not be added IMO.

Comment (lenary) [reply hw/top_earlgrey/dv/chip_sim_cfg.hjson:114]: @sriyerg Can you talk through the problem a bit more? I'm keen to understand the barriers to following this structure and approach.

This structure was presented as part of the [checklist](https://docs.opentitan.org/doc/project/checklist/#diftestsanity), and you seemed to agree at that point, including to the approach of starting with a hw-written test and then moving it to this proposed structure.

Comment (sriyerg) [reply hw/top_earlgrey/dv/chip_sim_cfg.hjson:114]: Should be:
```suggestion
      sw_test: sw/device/tests/dif_plic_sanitytest
```

The elf target is one level up.

**Pull request merged**